### PR TITLE
handle null assessment type when generating ICA and Summative student…

### PR DESCRIPTION
**Pull request by jtreuting**
… chunk reports

The problem was that the ExamQueryParams has a null assessment type when generating a student report for all types and now those queries need to have a type_id passed in.  We know the type when we are querying so we can add it easily before. 

Having issues with mockito that I could use some suggestions on, see below for more details.

### CODE REVIEW

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/stream/StudentReportProcessor.java [hunk 1]
```diff
@@ -123,13 +124,13 @@ public void generateStudentChunkReport(final Message<ProcessStudentChunkMessage>
 
         final ExamQueryParams queryParams = getQueryParams(payload.getReportId(), report.getReportRequest());
 
-        if (includeReports("iab", queryParams)) {
+        if (includeReports(AssessmentType.IAB.code(), queryParams)) {
```
Comment (jtreuting) [inline report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/stream/StudentReportProcessor.java]: unrelated but felt cleaner

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/stream/StudentReportProcessor.java [hunk 2]
```diff
@@ -190,13 +191,11 @@ private void appendIABReports(final Map<Long, StudentCompositeReport> compositeR
...
 
 
-        iabReports.forEach((key, value) -> compositeReports.computeIfPresent(key, (id, compositeReport) -> {
-                    return StudentCompositeReport.builder()
-                            .copy(compositeReport)
-                            .iabEla(value.get(ELA))
-                            .iabMath(value.get(MATH))
-                            .build();
-                }
+        iabReports.forEach((key, value) -> compositeReports.computeIfPresent(key, (id, compositeReport) -> StudentCompositeReport.builder()
```
Comment (jtreuting) [inline report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/stream/StudentReportProcessor.java]: no change just cleaner (suggested by IntelliJ)

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/stream/StudentReportProcessor.java [hunk 3]
```diff
@@ -205,11 +204,16 @@ private void appendICAReports(final Map<Long, StudentCompositeReport> compositeR
                                   final User user,
                                   final Collection<Long> studentIds,
                                   final ExamQueryParams examQueryParams) {
+        ExamQueryParams icaQueryParams = ExamQueryParams.builder()
+                .copy(examQueryParams)
+                .assessmentTypeCode(AssessmentType.ICA.code())
```
Comment (jtreuting) [inline report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/stream/StudentReportProcessor.java]: this is the fix, we are inside a private method `appendICAReports` so we know the type of assessment here and can add it using the builder().copy()

File: report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/model/ExamQueryParamsTest.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+                .gradeId(1)
+                .groupId(1L)
+                .schoolYear(1)
+                .studentId(1L)
+                .queryPermissionType(ExamQueryParams.PermissionType.Group)
+                .disableTransferAccess(true)
+                .build();
+        ExamQueryParams examQueryParams2 = ExamQueryParams.builder().copy(examQueryParams1).build();
+
+        assertThat(examQueryParams1).isEqualToComparingFieldByField(examQueryParams2);
```
Comment (jtreuting) [inline report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/model/ExamQueryParamsTest.java]: simple test for the new copy in the builder

File: report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/stream/StudentReportProcessorTest.java [hunk 5]
```diff
@@ -160,9 +160,9 @@ public void setup() throws Exception {
             return callback.doWithRetry(mock(RetryContext.class));
         }).when(retryTemplate).execute(any(), any(), any());
 
-        final byte[] pdfConetnt = "pdfContent".getBytes();
+        final byte[] pdfContent = "pdfContent".getBytes();
         when(htmlToPdfProcessor.process(eq(htmlContent), eq(printOptions)))
-                .thenAnswer(invocation -> new ByteArrayInputStream(pdfConetnt));
+                .thenAnswer(invocation -> new ByteArrayInputStream(pdfContent));
```
Comment (jtreuting) [inline report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/stream/StudentReportProcessorTest.java]: unrelated typo fix

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/stream/StudentReportProcessor.java [hunk 6]
```diff
@@ -223,11 +227,16 @@ private void appendSummativeReports(final Map<Long, StudentCompositeReport> comp
                                   final User user,
                                   final Collection<Long> studentIds,
                                   final ExamQueryParams examQueryParams) {
+        ExamQueryParams sumQueryParams = ExamQueryParams.builder()
```
Comment (wtritch) [inline report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/stream/StudentReportProcessor.java]: nit: final

File: report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/stream/StudentReportProcessorTest.java [hunk 7]
```diff
@@ -330,6 +330,14 @@ public void itShouldWritePdfContentsToTemporaryStorage() throws Exception {
         verify(temporaryStorage).writeChunk(eq(payload.getReportId()), eq(0), any(InputStream.class));
     }
 
+    @Test
+    public void itShouldHandleNullAssessmentTypeWhenGeneratingStudentChunk() {
+        when(icaReportRepository.findAllForStudentsByExamFilter(any(User.class), anyCollection(), argThat(p -> p.getAssessmentTypeCode() == null)))
+                .thenThrow(IllegalArgumentException.class);
+
+        writer.generateStudentChunkReport(message);
```
Comment (jtreuting) [inline report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/stream/StudentReportProcessorTest.java]: This works and the test passes with the fix and didn't before the fix, but I'm getting the following mockito error that I'm guessing will impact CI.

```
org.mockito.exceptions.misusing.UnnecessaryStubbingException: 
Unnecessary stubbings detected in test class: StudentReportProcessorTest
Clean & maintainable test code requires zero unnecessary code.
Following stubbings are unnecessary (click to navigate to relevant line of code):
  1. -> at org.opentestsystem.rdw.reporting.processor.stream.StudentReportProcessorTest.itShouldHandleNullAssessmentType(StudentReportProcessorTest.java:348)
Please remove unnecessary stubbings or use 'lenient' strictness. More info: javadoc for UnnecessaryStubbingException class.
```

In the setup we have a mock for this method already that accepts any ExamQueryParams and returns a HashMap 

In setup:
```
when(icaReportRepository.findAllForStudentsByExamFilter(any(User.class), anyCollection(), any(ExamQueryParams.class)))
                .thenReturn(new HashMap<>());
```

Based on this: https://stackoverflow.com/questions/2684630/how-can-i-make-a-method-return-an-argument-that-was-passed-to-it?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa I tried to make the return type conditional so there was only a single definition for this mock method but couldn't get it to work:

```
when(icaReportRepository.findAllForStudentsByExamFilter(any(User.class), anyCollection(), any(ExamQueryParams.class)))
                .thenReturn(new Answer<Map<Long, Map<Subject, IcaSummativeReport>>>() {
                    @Override
                    public Map<Long, Map<Subject, IcaSummativeReport>> answer(InvocationOnMock invocation) throws Throwable {
                        Object[] args = invocation.getArguments();

                        if (((ExamQueryParams)args[ 2 ]).getAssessmentTypeCode() == null)
                            throw new IllegalArgumentException();

                        return new HashMap<Long, Map<Subject, IcaSummativeReport>>();
                    }
                });
```

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/stream/StudentReportProcessor.java [hunk 8]
```diff
@@ -205,11 +204,16 @@ private void appendICAReports(final Map<Long, StudentCompositeReport> compositeR
                                   final User user,
                                   final Collection<Long> studentIds,
                                   final ExamQueryParams examQueryParams) {
+        ExamQueryParams icaQueryParams = ExamQueryParams.builder()
```
Comment (wtritch) [inline report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/stream/StudentReportProcessor.java]: nit: final

File: report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/stream/StudentReportProcessorTest.java (hunk 7 above)
Comment (wtritch) [reply report-processor/src/test/java/org/opentestsystem/rdw/reporting/processor/stream/StudentReportProcessorTest.java]: Hmm... rather than mocking out a failing case and assuring that it isn't hit, I'd probably confirm that it *is* doing what you want, adding something like:
```
        verify(icaReportRepository).findAllForStudentsByExamFilter(
                any(User.class),
                anyCollection(),
                examFilterCaptor.capture()
        );
        assertThat(examFilterCaptor.getValue().getAssessmentTypeCode()).isNotBlank();
```
to the existing repository interactions
